plenum/server/consensus/checkpoint_service.py

```diff
@@ -83,37 +77,16 @@ def process_checkpoint(self, msg: Checkpoint, sender: str) -> (bool, str):
             return None, None
         self._logger.info('{} processing checkpoint {} from {}'.format(self, msg, sender))
         result, reason = self._validator.validate(msg)
-        if result == PROCESS:
-            self._do_process_checkpoint(msg, sender)
-        return result, reason
+        if result != PROCESS:
+            return result, reason
 
-    def _do_process_checkpoint(self, msg: Checkpoint, sender: str) -> bool:
-        """
-        Process checkpoint messages
-
-        :return: whether processed (True) or stashed (False)
-        """
-        seqNoEnd = msg.seqNoEnd
-        seqNoStart = msg.seqNoStart
-        key = (seqNoStart, seqNoEnd)
-
-        if key not in self._checkpoint_state or not self._checkpoint_state[key].digest:
-            self._stash_checkpoint(msg, sender)
-            self._remove_stashed_checkpoints(self.last_ordered_3pc)
-            self._start_catchup_if_needed()
-            return False
+        key = self.CheckpointKey(view_no=msg.viewNo,
```

Comment:
It wasn't so obvious before, but there was quite essential line:
`self._remove_stashed_checkpoints(self.last_ordered_3pc)` in `_do_process_checkpoint`. It was needed to avoid unnecessary catchup starts after unstashing of stashed checkpoints after catchup. This can be a reason of a failing integration test.

Reply:
Reimplemented this functionality by adding more checks into catchup triggering logic. It indeed fixed failing integration test